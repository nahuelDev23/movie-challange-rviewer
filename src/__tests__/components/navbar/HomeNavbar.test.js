import { shallow } from 'enzyme'
import { HomeNavbar } from '../../../components/navbar/HomeNavbar';

describe('Pruebas en <HomeNavbar/>', () => {
    test('Debe mostrar <HomeNavbar/> correctamente', () => {
        const wrapper = shallow(<HomeNavbar />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe contener solo una etiqueta nav',()=>{
        const wrapper = shallow(<HomeNavbar />);
        expect(wrapper.find('nav').length).toBe(1);
    })

    test('Debe haber un link que al hacer click re direccione al home y otros 2 que no vayan a ningun lado',()=>{
        const wrapper = shallow(<HomeNavbar />);
        const link = wrapper.find('Link');
        
        expect(link.at(0).containsMatchingElement( <img  alt="Logo Dreadful Tomato" />)).toBe(true);
        expect(link.at(0).props().to).toBe('/');
        expect(link.at(1).props().to).toBe('#');
        expect(link.at(2).props().to).toBe('#');
    })
})