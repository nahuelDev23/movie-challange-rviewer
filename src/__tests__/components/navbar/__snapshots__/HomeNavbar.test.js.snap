// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeNavbar/> Debe mostrar <HomeNavbar/> correctamente 1`] = `
<nav
  className="nav"
>
  <div
    className="nav__logo"
  >
    <Link
      to="/"
    >
       
      <img
        alt="Logo Dreadful Tomato"
        src="logo.png"
      />
    </Link>
  </div>
  <div
    className="nav__actions"
  >
    <div
      className="nav__login"
    >
      <p>
        <Link
          to="#"
        >
          Login
        </Link>
      </p>
      <img
        alt="icono para el login"
        src="icon-login.png"
      />
    </div>
    <div
      className="nav__trial btn btn--active"
    >
      <Link
        to="#"
      >
         Start your free trial
      </Link>
    </div>
  </div>
</nav>
`;
